src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWallet.kt

```diff
@@ -0,0 +1,119 @@
+package fr.acinq.lightning.blockchain.electrum
+
+import fr.acinq.bitcoin.*
+import fr.acinq.lightning.utils.Connection
+import fr.acinq.lightning.utils.lightningLogger
+import fr.acinq.lightning.utils.sat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlin.native.concurrent.ThreadLocal
+
+data class WalletState(val addresses: Map<String, List<UnspentItem>>) {
+    val utxos: List<UnspentItem> = addresses.flatMap { it.value }
+    val balance: Satoshi = utxos.sumOf { it.value }.sat
+}
+
+private sealed interface WalletCommand {
+    companion object {
+        object ElectrumConnected : WalletCommand
+        data class ElectrumNotification(val msg: ElectrumResponse) : WalletCommand
+        data class AddAddress(val bitcoinAddress: String) : WalletCommand
+    }
+}
+
+
+/**
+ * A very simple wallet that only watches one address and publishes its utxos.
+ */
+class ElectrumMiniWallet(
+    val chainHash: ByteVector32,
+    private val client: ElectrumClient,
+    private val scope: CoroutineScope
+) : CoroutineScope by scope {
+
+    // state flow with the current balance
+    private val _walletStateFlow = MutableStateFlow(WalletState(emptyMap()))
+    val walletStateFlow get() = _walletStateFlow.asStateFlow()
+
+    // all currently watched bitcoin addresses and their corresponding script hash
+    private var scriptHashes: Map<ByteVector32, String> = emptyMap()
```

Comment:
Should we track addresses or pubkeyscripts internally? The former is more human friendly, the latter is more generic.